blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/BlueOceanRootAction.java

```diff
@@ -52,20 +59,37 @@ public Object getTarget() {
 
         StaplerRequest request = Stapler.getCurrentRequest();
 
-        if(enableJWT && request.getOriginalRestOfPath().startsWith("/rest/")) {
-            Authentication tokenAuthentication = JwtAuthenticationToken.create(request);
+        if(request.getOriginalRestOfPath().startsWith("/rest/")) {
+            if(enableJWT) {
+                Authentication tokenAuthentication = JwtAuthenticationToken.create(request);
+
+                //create a new context and set it to holder to not clobber existing context
+                SecurityContext securityContext = new SecurityContextImpl();
+                securityContext.setAuthentication(tokenAuthentication);
+                SecurityContextHolder.setContext(securityContext);
+
+                //TODO: implement this as filter, see PluginServletFilter to clear the context
+            } else {
+                // Add X-Blueocean-Refresher to response, so that we can detect when the user changes. Wont be needed with JWT
+                MessageDigest md = null;
+                try {
+                    md = MessageDigest.getInstance("SHA-1");
```

Comment:
yeah this is a bit absurd isn't it, is there no way to statically pick one that is known to be installed? 

cc @i386 is there a standard jenkins way of hashing (I heard you mentioned)